### PR TITLE
Integration Spec

### PR DESCRIPTION
**Pull request by binyamolango**
# Hi Reviewer 👋

## Here are the changes I and my partner implemented to the project

Used Capybara to write integration tests for each view in my project.

Wrote the following tests for the user index page:

- Display the username of all other users.
- Display the profile picture for each user.
- Display the number of posts each user has written.
- Clicking on a user will redirect to that user's show page.

Wrote the following tests for the user show page:

- Display the user's profile picture.
- Display the user's username.
- Display the number of posts the user has written.
- Display the user's bio.
- Display the user's first 3 posts.
- Display a button that lets me view all of a user's posts.
- Clicking a user's post will redirect me to that post's show page.
- Clicking to see all posts will redirect me to the user's post's index page.

Wrote the following tests for the user post index page:

- Display the user's profile picture.
- Display the user's username.
- Display the number of posts the user has written.
- Display a post's title.
- Display some of the post's body.
- Display the first comments on a post.
- Display how many comments a post has.
- Display how many likes a post has.
- Display a button for adding new post.
- Clicking on a post will redirect me to that post's show page.

Wrote the following tests for the post-show page:

- Display the post's title.
- Display who wrote the post.
- Display how many comments it has.
- Display how many likes it has.
- Display the post body.
- Display the username of each comment.
- Display the comment each comment left.

Included integration specs for other features I have implemented like adding new posts, comments, and likes.

Screenshot 2023-11-28 161043
Screenshot 2023-11-28 161135

### CODE REVIEW

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @binyamolango, and @BilalLiaquat7👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job fixing N+1 problem. ✔️ 
- Good job with integration specs. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] Please make sure to add screenshots of your app's logs in the PR description before and after fixing the N+1 problem as requested in the project requirements.

- [ ] All of the integration tests are failing. Please make sure to fix this and all tests should pass without any errors.
image


- [ ] Posts index page missing the below test. Please make sure to add it and it should pass without any errors as requested in the project requirements.
  - `Tests that asserts you can see a section for pagination if there are more posts than fit on the view.`

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by binyamolango** [comment]
Hi @thecodechaser, Thank you for the insightful suggestions. On my side all the tests are passing. I don't why it is failing on your side. I attached the screenshot here.

Screenshot 2023-11-28 161625

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -48,20 +48,14 @@
...
     expect(current_path).to eq(user_post_path(@user1, @user1.posts[0]))
   end
 
-  scenario 'clicking on a Add new post redirects to the post new page' do
+  scenario 'display pagination button' do
     visit user_posts_path(@user1)
-    click_link('Add new post')
-    expect(current_path).to eq(new_user_post_path(@user1))
+    expect(page).to have_button('Pagination')
   end
```
Comment (TadesseDev) [inline spec/views/post_index_spec.rb:60]: - [ ] Dear @binyamolango, I see you have made changes to your test as suggested by the previous reviewer. But you have changed a button name and test the button with a name `pagination`, while in reality what we want to test is `Tests that asserts you can see a section for pagination if there are more posts than fit on the view.`. we still ned the `add new post` button, yet we wanna assert if the page cannot hold all the poste's it should show a paggenation.

Comment (TadesseDev) [inline spec/views/post_index_spec.rb]: - [x] Dear @binyamolango, I see you have made changes to your test as suggested by the previous reviewer. But you have changed a button name and test the button with a name `pagination`, while in reality what we want to test is `Tests that asserts you can see a section for pagination if there are more posts than fit on the view.`. we still ned the `add new post` button, yet we wanna assert if the page cannot hold all the poste's it should show a paggenation.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @binyamolango 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

199733072-bdacca19-8df9-43a7-be93-3aff2f1c15c8

 ## Highlights
 - [x] Good job making most of the changes requested by the previous reviewer. 🙏


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ghost — APPROVED** [review]
Hi @binyamolango , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Pagination property is tested well 👍 
✅ Polished and Professional documentation👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**